BaseTools/Plugin/OverrideValidation/OverrideValidation.py

```diff
@@ -687,28 +722,50 @@ def path_parse():
     else:
         dummy_list = []
         pathtool = Edk2Path(Paths.WorkSpace, dummy_list)
+        if Paths.DeprecatedPath is not None:
+            # Generate deprecation warning line
+            #    DEPRECATION_FORMAT_VERSION_1 = (1, 4)   # Version 1: # DEPRECATED : VERSION | PATH_TO_NEW_MODULE_TO_USE | YYYY-MM-DDThh-mm-ss
+
+            if Paths.DeprecationReplacementPath is not None:
+                pkg_path = pathtool.GetContainingPackage(Paths.DeprecationReplacementPath)
+                if pkg_path is not None:
+                    rel_path = Paths.DeprecationReplacementPath[Paths.DeprecationReplacementPath.find(pkg_path):]
+                else:
+                    rel_path = pathtool.GetEdk2RelativePathFromAbsolutePath(Paths.DeprecationReplacementPath)
+                    if not rel_path:
+                        print(f"{Paths.DeprecationReplacementPath} is invalid for this workspace.")
+                        sys.exit(1)
+                rel_path = rel_path.replace('\\', '/')
+            else:
+                rel_path = "REMOVED_COMPLETELY"
+
+            print("Copy and paste the following line(s) to your deprecated inf file(s):\n")
+            print('#%s : %08d | %s | %s \n' % ('Deprecated', DEPRECATION_FORMAT_VERSION_1[0], rel_path, datetime.now(timezone.utc).strftime("%Y-%m-%dT%H-%M-%S")))

```

Comment:
I remember our team had some discussions about this internally. Would it be a better approach to make it like "time-of-generation | grace period". Then before the grace period, we can just print warning. After grace period, we break the build?

@spbrogan @makubacki  thoughts?